BDArmory/Control/BDModuleOrbitalAI.cs

```diff
@@ -983,6 +1179,90 @@ public static float GetMaxThrust(Vessel v)
             thrust += VesselModuleRegistry.GetModules<ModuleRCS>(v).Where(rcs => rcs != null && rcs.useThrottle).Sum(rcs => rcs.thrusterPower);
             return thrust;
         }
+
+        private static float GetConsumptionRate(Vessel v)
+        {
+            float consumptionRate = 0.0f;
+            foreach (var engine in VesselModuleRegistry.GetModuleEngines(v))
+                consumptionRate += Mathf.Lerp(engine.minFuelFlow, engine.maxFuelFlow, 0.01f * engine.thrustPercentage) * engine.flowMultiplier;
+            return consumptionRate;
+        }
+        
+        //Controller Integral
+        Vector3 directionIntegral;
+        float pitchIntegral;
+        float yawIntegral;
+        float rollIntegral;
+        Vector3 prevTargetDir;
+        void AttitudeControl(FlightCtrlState s)
+        {
+            Vector3 targetDirection = fc.attitude;
+            Vector3 currentRoll = -vesselTransform.forward;
+            Vector3 rollTarget = currentRoll;
+            if (targetVessel != null) // If we have a target, put broadside toward target
```

Comment:
"Broadside" needs a control setting in the AI as it depends on ship design.
If you put your main weapons all on one side of the craft (e.g., on the front, or mounted dorsally), then you want that side facing the target.
On the other hand, if the main guns are mounted symmetrically laterally, then you want to roll such that both sides can fire at the target (i.e., the roll target would be `±vesselTransform.forward`, whichever is closer) and vice-versa for dorsally/ventrally mounted main weapons.